test/analytics/tests/analytics_from_the_server.ts

```diff
@@ -58,87 +56,120 @@ export default function ({ getService }: FtrProviderContext) {
         },
       ]);
     });
+    describe('after setting opt-in', () => {
+      let actions: Action[];
+      let context: Action['meta'];
 
-    it('after setting opt-in, it should extend the contexts and send the events', async () => {
-      await ebtServerHelper.setOptIn(true);
+      before(async () => {
+        await ebtServerHelper.setOptIn(true);
+        actions = await getActions();
+        context = actions[1].meta;
+      });
 
-      const actions = await getActions(3);
+      it('it should extend the contexts and send the events', async () => {
+        // Validating the remote PID because that's the only field that it's added by the FTR plugin.
+        // const context = actions[1].meta;
+        expect(context).to.have.property('pid');
+        expect(context.pid).to.be.a('number');
 
-      // Validating the remote PID because that's the only field that it's added by the FTR plugin.
-      const context = actions[1].meta;
-      expect(context).to.have.property('pid');
-      expect(context.pid).to.be.a('number');
+        // Some context providers emit very early. We are OK with that.
+        expect(context).to.have.property('pid');
+        expect(context.pid).to.be.a('number');
 
-      // Some context providers emit very early. We are OK with that.
-      const initialContext = actions[2].meta[0].context;
+        const [optInAction, extendContextAction] = actions;
+        expect(optInAction).to.eql({ action: 'optIn', meta: true });
+        expect(extendContextAction).to.eql({ action: 'extendContext', meta: context });
+      });
 
-      const reportEventContext = actions[2].meta[1].context;
-      expect(context).to.have.property('pid');
-      expect(context.pid).to.be.a('number');
+      it('should send events', async () => {
```

Comment:
nit: I think we can split this test:

1. Initial calls to `reportEvents` from cached events group the requests by event_type
2. `test-plugin-lifecycle` is received in the correct order (setup then start)